client/cluster.go

```diff
@@ -702,7 +702,10 @@ func (c *ClusterClient) terraformRemoteWorkspaceApply() error {
 	}
 
 	if err := c.tfe.Init(); err != nil {
-		return fmt.Errorf("error initializing TFE workspace: %w", err)
+		c.logger.Warn("Initialization of terraform remote workspace failed, retrying...")
```

Comment:
I think it would be more nice if we could add a `TLSTimeoutError` that the init function could return so that  we can retry only if that's the error we're getting.

Reply:
Nice! I will try to put together something like that :slightly_smiling_face: 

Regarding where we get this error, I think it has only ever happened during init, but I'm not completely sure. If anyone knows more about it I would be very happy to hear it!

Reply:
At least this exoscale pipeline https://github.com/elastisys/ck8s-pipelines/runs/1201625845?check_suite_focus=true has this error on destroy.

Reply:
Ok. Does it always happen för the TFE or is it also for other?

Reply:
Good question. As far as I can find, it looks like its only TFE.